### ISSUE TITLE
F-Droid listing

### ISSUE THREAD

**Issue by IzzySoft** [helcel-net/beans#2]
I just noticed your Readme links to F-Droid, but that link gives a 404 as the app is not there. Maybe you instead want to [pick a badge](https://gitlab.com/IzzyOnDroid/repo/#if-my-app-is-available-here-do-you-have-a-badge-i-can-use-to-link-to-it) to link to the IzzyOnDroid repo, where [your app is listed](https://apt.izzysoft.de/packages/net.helcel.beans)? :wink:

Btw: if you'd replace those "graphical bullet points" in your fastlane's `full_description.txt` with asterisks (`*`), that file would render fine as Markdown and thus format much better :wink:

**Comment by IzzySoft** [comment]
One more thing:

At IzzyOnDroid we support [Reproducible Builds](https://reproducible-builds.org/) (see: [Reproducible Builds, special client support and more at IzzyOnDroid](https://android.izzysoft.de/articles/named/iod-rbs-mirrors-clients)). Trying for yours, I was able to successfully generate the APK using `./gradlew assembleRelease`, but the resulting APKs were not identical. One file causing differences is `res/M7.json`, belonging to aboutLibraries and including a generated timestamp. To avoid that, you could add the following to your `build.gradle`:

```gradle
aboutLibraries {
    // Remove the "generated" timestamp to allow for reproducible builds
    excludeFields = ["generated"]
}
```

I'm not yet sure about the other differences, which are in your `classes.dex`:

```diff
 magic               : 'dex\n039\0'
-checksum            : d8bd11ae
-signature           : 230a...9780
+checksum            : 123b108b
+signature           : 8288...9622
 file_size           : 1851372
```

They could just indicate a non-deterministic build – but we can verify that once the above was fixed.

We'd appreciate if you could help making your build reproducible. We've prepared some [hints on reproducible builds](https://gitlab.com/IzzyOnDroid/repo/-/wikis/Reproducible-Builds) for that.

Looking forward to your reply!

**Comment by IzzySoft** [comment]
Still anyone alive here (apart from the renovate-bot)?

**Comment by choelzl** [comment]
Yes life keeping me a bit busy will try to resolve these issues and things over the next days

**Comment by IzzySoft** [comment]
Thanks!

**Comment by choelzl** [comment]
I tried making the change for "aboutLibraries" and made a release 2 days ago but it hasnt been picked up yet. Anything i need to do to enable reproducible builds on my end ? Is there a reason the release is not showing on izzy ?

**Comment by IzzySoft** [comment]
> Is there a reason the release is not showing on izzy ?

Most likely your tag naming:

```
$ iod repo get net.helcel.beans
net.helcel.beans: looking for 'https://api.github.com/repos/helcel-net/beans/releases'
net.helcel.beans: checking tag '1.0c'
net.helcel.beans: lastRelNo set to '1.0c', checking for files
net.helcel.beans: Upstream file date (2024-11-09 12:45) is newer than ours (2024-07-08 01:40).
net.helcel.beans: returning ['1.0c','https://github.com/helcel-net/beans/releases/download/1.0c/app-signedRelease.apk',1731152725]
net.helcel.beans: 1.0 up-to-date.
```

`aa` & `c` are not following [semantic versioning](https://semver.org), so what remains is that each tag represents version 1.0, which is already here :man_shrugging: Moreover, `1.0a` _would_ match semver and represent the alpha. Which would not have been picked up if the rc1 (released before) had already been here, as "rc" is considered "newer" than an alpha with the same version.

Further, it seems your versioning isn't even in sync: your `build.gradle` says `1.0b` (but at least has the `versionCode` increased to `2` – while there are 4 releases (suggesting `versionCode` should be 4 or higher) with no 1.0b ("v1.0 beta" according to semver) at all.

I could pull in your "1.0aa" manually (which I have done now), but I'd strongly suggest fixing your versioning for the next releases please – as I cannot keep track of doing that manually on each of your releases :wink:

> I tried making the change for "aboutLibraries" 

That was successful: I could reproduce the APK – but only when calling `assembleSignedRelease` with the signing call removed (`assembleRelease` does not produce an identical APK as the flavor name used seems to have effect on the hash of the `classes.dex`). No prob, I can set up the recipe accordingly (so no changes to that part required – more important to keep it consistent, so the recipe does not need to be updated manually on future releases).

**TL;DR:** 1.0c has been pulled in manually now and is RB – but please use proper versioning from now on (and remember to always increase `versionCode`).

Thanks!

**Comment by IzzySoft** [comment]
![image](https://github.com/user-attachments/assets/da8cb5e1-862d-4c26-bfff-85543d6072f3)

[Welcome toot](https://floss.social/@IzzyOnDroid/113464204401616635):

![image](https://github.com/user-attachments/assets/cc26ed25-0217-4406-b7be-40285ac9cd84)

**Comment by choelzl** [comment]
>  [semantic versioning](https://semver.org)

Ok i will fix my version numbering in future releases. Thank you the reproducible build assistance 🎉

**Comment by IzzySoft** [comment]
Gladly – and thanks from us as well! :heart_eyes:
